yt_dlp/extractor/nbc.py

```diff
@@ -736,7 +736,7 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, video_id)
 
         nbc_data = self._search_json(
-            r'<script>\s*var\s+nbc\s*=', webpage, 'NBC JSON data', video_id)
+            r'<script>(?:\s*var\s+nbc\s*=|.+?Object\.assign\(nbc,\s*)', webpage, 'NBC JSON data', video_id)
```

Comment:
I'd prefer to avoid the `.+?` and do this
```suggestion
            r'(?:<script>?:\s*var\s+nbc\s*=|Object\.assign\(nbc,)', webpage, 'NBC JSON data', video_id)
```

Reply:
ready for review